####Opening BLAST program

In [1]:
!/Applications/ncbi-blast-2.5.0+/bin/blastp

BLAST query/options error: Either a BLAST database or subject sequence(s) must be specified
Please refer to the BLAST+ user manual.


####Viewing geoduck data
Steven & Brent [sampled geoduck in 2015](http://onsnetwork.org/halfshell/2015/08/11/big-day-big-clam/); this is me getting a glimpse of the raw data of said _p. generosa_ genome; BGI did the sequencing, and the files are [here](http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/). 

In [4]:
! curl http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq \
| head 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTT

#####Prepping directory folders

In [1]:
pwd

'/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter'

In [2]:
cd ..

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project


In [3]:
ls

Jupyter/


In [6]:
mkdir data/ analyses/ scripts/

mkdir: data/: File exists


In [7]:
ls

Jupyter/  analyses/ data/     scripts/


In [8]:
cd data/

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


####Downloading geoduck genome file into local directory

In [10]:
!wget http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq

/bin/sh: wget: command not found


_**Interesting note: the command `wget` is not installed on Macs**_

In [15]:
!curl -O http://owl.fish.washington.edu/P_generosa_genome_assemblies_BGI/20160512/Panopea_generosa.scafSeq.zip

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  350M  100  350M    0     0   898k      0  0:06:40  0:06:40 --:--:--  909k


**Notes:**  
* Remember to use `!` before all commands (except for `cd` `pwd` `ls` etc.?)  
* -o (lowercase o) the result will be saved in the filename provided in the command line  
* -O (uppercase O) the filename in the URL will be taken and it will be used as the filename to store the result
* Helpful `curl` tips: http://www.thegeekstuff.com/2012/04/curl-examples/

In [32]:
ls

Panopea_generosa.scafSeq      Panopea_generosa.scafSeq.zip


In [48]:
!head Panopea_generosa.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATCATGTACCATTAAATCTGATA
CTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgtttttgttgttttttcagcctgaat


In [49]:
!tail Panopea_generosa.scafSeq

GAGGACAAAATTGAGATTGTTGTGGACGGGAAACAGGAAACATATGAAGACGAGTTCGCGGTCATTAAGGTTAAGGTTTCAGACTTGGACGAAATGAAAA
TACCTTATGAGTTAGTTTAACCGCGGTACGCCAATGTATGATATACTTATACCAGTAAATGATTAGGAGGAATATTAATGAGCGAGAAAGTAGAATTGAT
TGTTGTTGATATGGTTGACGAGGAAATTAGTACTAAGTTCCAATTGTGTCCTGAAGACTACTCGAAAGTGTATACACTGCAGTTGTTCAAACAGGACTAT
AAAAAGGACACCAAAGAATGGGTCAGTGATGCGGAAACCACCAAACAATACGATGCGGATATGAAAGAGTTGGGTAACCCTAGCGTGGGTGATAAGGTCA
ATGTGTTCGTTTCAAACAACGGCAAAGCGTATTTGACTGAAAAAACATTCGTGGAAACGGTGAAACCTAGCGGGAAAGATGCGAACAAATTGTTCACAAA
TTGCGAAATTGAATCGTTCCATGATTCGCCAAAAGGGCGACGCGTGATTGTCAAGTATAAAAAGAATTTTTACGAGTTTAGTTTCAATTATTCCGTTTGG
ATTCCGTCAATTGAGAAGTACGTTCCAAATGATGCCAAAATGGAAAAGGCAAAAGTGCGATTCAACGATATTTTTGAACATTCGTTGATTACGTGGGAAA
ACGCACTAGAAACGGACGCGGATGGTGTTAGCATCCATAAGGGTGTTAAAATGAACTGCTTGGTTCGCAAAAACGAATTGGACGCAACTGGTCAAACAGG
TTGGTTGGAACCGTTGAAAATTGACGAACCAGATGTTGACGAAGACGACCTTCCGTTCTAAAAATAGTTACAAAAAAGTGTAAGTTTAGGCTTGCACTTT
TTATTTTACCGTGGTAAGATATACTCAACAAGTTGAGTGA


**Unsure how to navigate in the `less` program**

In [50]:
!less Panopea_generosa.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCT CATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatg ctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagtta cccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTT GTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTG CAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCT TAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaat tgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATC ATGTACCATTAAATCTGATA
CTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgttttt gttgttttttcagcctgaat
atagataactataacATGTTACTAGGAATGGGGTTCCCTAAGGGAGCTTCAGAAGAG

**Removing first row (metadata), viewing only first 5 rows of sequences**

In [59]:
!tail -n +2 Panopea_generosa.scafSeq | head -n 5

ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
tail: stdout: Broken pipe


**Experimenting with `grep` and color coding**

In [66]:
!tail -n +2 Panopea_generosa.scafSeq | head -n 5 \
| grep -C2 "AAATACA" --color=auto

ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
tail: stdout: Broken pipe


In [67]:
!file Panopea_generosa.scafSeq

Panopea_generosa.scafSeq: ASCII text


**Trying to really manipulate the genome data in order to Blast in comparison to transcriptome data. First let's see the first 20 rows of the data:**

In [1]:
pwd

'/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/Jupyter'

In [4]:
cd ..

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project


In [5]:
cd data

/Users/shlaura3/Documents/SAFS/FISH_546_Bioinformatics/546-Bioinformatics/2016-10_Geo-Ann-Project/data


In [6]:
ls

Panopea_generosa.scafSeq      Panopea_generosa.scafSeq.zip


In [19]:
!head -n 60 Panopea_generosa.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgct
taaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATA
CATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccat
ctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATAT
ACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAG
GCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGAC
ACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcact
cgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATCATGTACCATTAAATCTGATA
CTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgtttttgttgttttttcagcctgaat
atagataactataacATGTTACTAGGAATGGGGTTCCCTAAGGGAGCTTCAGAAGAGGCACTCAGACAGGGAAAC

**So, I take it the scaffolds are very long. I'm curious what "26.6" means -> is this the scaffold #? I'll try to see how many scaffolds there are...**

In [9]:
! grep -c "^>" Panopea_generosa.scafSeq

1296135


**Ok, there are nearly 1.3 million scaffolds. Steven suggested paring this data down to only very large scaffolds, such as on those >70k. This script was borrowed from [Steven's notebbok](https://sr320.github.io/student-fish546-2016/data-description/)**

In [10]:
!awk '!/^>/ { next } { getline seq } length(seq) >= 70000 { print $0 "\n" seq }' \
../data/Panopea_generosa.scafSeq > ../data/Panopea_generosa_scaff-70k.scafSeq

In [11]:
ls

Panopea_generosa.scafSeq            Panopea_generosa_scaff-70k.scafSeq
Panopea_generosa.scafSeq.zip


In [16]:
! wc Panopea_generosa_scaff-70k.scafSeq

       0       0       0 Panopea_generosa_scaff-70k.scafSeq


**Hmm... the new file doesn't contain anything; turns out this is because there are "hard line-breaks" within scaffolds, so the search isn't finding any super long contigs. Now using a script (from Steven) to remove these line breaks:**

In [20]:
!awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" } END { printf "%s", n }' \
../data/Panopea_generosa.scafSeq > ../data/Panopea_generosa_No-Line-Breaks.scafSeq

In [22]:
!head -n 20 ../data/Panopea_generosa_No-Line-Breaks.scafSeq

>scaffold1 26.6
ACTAACAGACGATCATTAAATACAGTCATACAGAGCAAATGAATTAAGCAATAACCTTGTCTTTAAAGGCCACTTTTGCTCATTCCCTtgtcttgctgcttaaaacaagcttgactgtagatgtatccttaaaggctctcacaaagcattataaccgttatatcaattcattctatgatgctttaGCCCTTTACTCCATACATACGTGTTTTCCCGCACCAATCACAGAAtgcctttagaaactgttttccactctgtcagttaatataaacataagttacccaataattcaaaagccatctatattagaggtgtccaaaggcgtctctattaggtgtccaaaggcgtcgatatgaagtaaagggttaaatcaaTTTTTTGTAACCCCATCTGTTAATATACATATATCTAGCTGATGTAATTTACTATGACCTTAAAATGATTTTCGTAAACTCAGTAAAGCTCATTTTAACAAGTTTGCAAAATATATATTATGCAAGGCTGTGTCGGGCATTATATCATTTTTTGCCAATGAGTTGAATAAATTTATCATTGACTTTAAAAAGGTTTCCATCTATCTTAGAAATAGATATGACGGACACTATGGTTTTACCATGACAAGTCAACaaatcgttatcatgtaccattaaatctgatactatattggttttccaggaaattgttttaattccatggcactcgtaaatattCACTTGTTTTCTGATGTAACTTGTTTTTGTTGTTTTTTCAGCCTGAATATAGATAACTATAACNGTTATCATGTACCATTAAATCTGATACTATATTGGTTTTCCAGGAAATTGTTTTAATTCCATGGCACTCGTAAatattcacttgttttctgatgtaacttgtttttgttgttttttcagcctgaatatagataactataacATGTTACTAGGAATGGGGTTCCCTAAGGGAGCTTCAGAAGAGGCACTCAGACAGGGAAACAATGATGTC

**Looks like ALL line breaks have gone away now, b/c when I ask for the first 20 lines it produces many more. I'll try the >70k script again, see what happens**

In [23]:
!awk '!/^>/ { next } { getline seq } length(seq) >= 70000 { print $0 "\n" seq }' \
../data/Panopea_generosa_No-Line-Breaks.scafSeq > ../data/Panopea_generosa_scaff-70k.scafSeq

In [25]:
! grep -c "^>" ../data/Panopea_generosa_scaff-70k.scafSeq

20


**Great! I've subselected the largest 20 scaffolds (from ~1.3M)**